### DISCUSSION TITLE
how to build autoconf project?

### DISCUSSION THREAD

**Discussion by stsp** [mesonbuild/meson#12606] [Q&A]
I do this:
```
p = mod.add_project('configure',
  configure_options : [
    '--prefix=@PREFIX@'
  ],
  verbose: true
)
```
configure is running OK.

But when I do
```
$ meson compile --verbose
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -v
ninja: no work to do.
```
... the build isn't running.

I can add this:
```
run_command(
  'make', '-C', meson.current_build_dir() / 'build',
  check: true,
  verbose: true
)
```
But in this case it is running `make`
directly during the setup stage, and
completely silently. I don't suppose
this is an intended use.

What should I do?

**Comment by eli-schwartz** [comment]
Did you add a `p.dependency('some-libname')` as a dependency to meson targets that are built by default?

**Reply by stsp** [reply]
No, I have no idea what `some-libname`
is. I build the tool, binary program.
Strictly speaking, modified nasm version.
I provide it via PPA, but I'd like to have
the meson-based wrap as a fall-back.

**Reply by eli-schwartz** [reply]
https://mesonbuild.com/External-Project-module.html

> Known limitations: Executables from external projects cannot be used uninstalled, because they would need its libraries to be installed in the final location. This is why there is no find_program() method.

**Reply by stsp** [reply]
I don't require uninstalled.
How to build & install? :)

**Reply by eli-schwartz** [reply]
cc @xclaesse 

It seems to not build by default, unless something actually links to it. If it does get built, it is installed as well, even if you manually built it.

**Reply by stsp** [reply]
Would be good to get that fixed.
In a mean time I added an explicit
run_command() for building, and
find_program() to locate the resulting
binary.
But I am not sure what to do next.
Will it play nicely with toolchain setup
overrides? What should I write in
`[binaries]`?

**Reply by stsp** [reply]
Or perhaps I should actually also
add explicit `make install` to avoid
any confusion.

**Reply by stsp** [reply]
`make install` via `run_command()`
looks problematic. It needs sudo...

> If it does get built, it is installed as well, even if you manually built it.

What exactly do you mean?
I can't seem to be able to manually
build it with anything else than
`run_command()`. And that one
doesn't install.

**Reply by eli-schwartz** [reply]
If you run `ninja <TAB>` it will offer autocompletion on targets, one of them will be a `subdir.stamp`.

So given an external project in `some_dep/meson.build` you would run `ninja some_dep/some_dep.stamp` to manually build it.
